### ISSUE TITLE
Improve automatic formatting

### ISSUE THREAD

**Issue by Nick-Mazuk** [finale-lua/lua-scripts#168]
Creating a new issue based on a conversation in #167. Here's the original message by @ThistleSifter:

---

I didn't run into any issues setting it up, but I have noticed that occasionally the style fixer does odd things.

Eg The comment in line 242 in `mixin/FCMCtrlListBox.lua` applies to the `else` below, so it should only be indented once, not twice.

I'm not a fan of some of the line splitting, although I understand that these boil down to personal preference so I'm not going to quibble about it:
- Line 447 in `mixin/FCMCtrlListBox.lua`, which it won't let me split into multiple lines so it visually matches the following arguments.
- I'd prefer to have the `for...do` in line 216 in `mixin/FCMCusomWindow.lua` be all on the same line
- The splitting on line 32 and below in `mixin/FCXCustomLuaWindow.lua` isn't great.

_Originally posted by @ThistleSifter in https://github.com/finale-lua/lua-scripts/issues/167#issuecomment-1114501284_

**Comment by Nick-Mazuk** [comment]
@rpatters1's reply:

---

I tend to agree with @ThistleSifter that I don't like the forced line splitting. In fact, I continue to do most of my editing in ZeroBrane because of the debugging. But it also allows me to fix lint issues found by VSCode without having to take the line splitting on board.

**Comment by Nick-Mazuk** [comment]
Yeah, this isn't the greatest formatter out there. I think having consistent, automatic formatting is more important than good formatting, but I'm totally open to improving the automated formatting.

Feel free to change the format config at https://github.com/finale-lua/lua-scripts/blob/master/luaconfig.config to something that looks better. Or if you find a better formatter that works with VS Code, we can switch to that.

Though any changes in that formatter should go in a separate PR so we can also reformat the entire repo in that PR. And if/when such a PR is created, we should merge it quickly to reduce merge conflicts with future PRs.

**Comment by rpatters1** [comment]
I'm going to piggyback this issue to mention that every time I launch Code it prompts me with some options about the "OpenResty" workspace, none of which is the one I want: to shut and go away for good. I have no idea what's going on, but if I allow it to modify my settings it switches me to LuaJIT, among other things.

**Comment by Nick-Mazuk** [comment]
Interesting. Is that every time you launch this repo, the `/src` folder of the repo, or every time you launch VS Code in general?

**Comment by Nick-Mazuk** [comment]
I don't have these extensions:

- Lua Debugger (devCAT)
- vscode-lua (trixnz)

The other two I have.

**Comment by ThistleSifter** [comment]
I'd be okay with removing the limit on line length. Other than that, I don't have any major issues with the formatter.

**Comment by rpatters1** [comment]
I concur with removing the limit on line length. The other formatting preferences are fine.

**Comment by Nick-Mazuk** [comment]
I'm fine with that. I'm not sure if there's a way to disable the line length feature, but you can set it to 9999. I'll approve any PRs changing this.

**Comment by rpatters1** [comment]
Does anyone know how to turn off line formatting completely? I expanded the line length to a huge number, but now I can't break of the lines manually.

**Comment by Nick-Mazuk** [comment]
It looks like you potentially can't turn it off. The code ensures the value exists and that it's an integer. I also tried using `-1`, but that seemed to disable formatting altogether, not just the line length.

https://github.com/Koihik/LuaFormatter/blob/0ef3575f3695b6553eaede2f7bd6a493ff0c520c/src/Config.cpp#L134

That's unfortunate.

**Comment by ThistleSifter** [comment]
I guess all we can do is hope that respecting manual line breaks is added someday: https://github.com/Koihik/LuaFormatter/issues/240
